cartographer_ros/cartographer_ros/assets_writer_main.cc

```diff
@@ -60,68 +62,67 @@ namespace carto = ::cartographer;
 void Run(const string& trajectory_filename, const string& bag_filename,
          const string& configuration_directory,
          const string& configuration_basename, const string& urdf_filename) {
-  std::ifstream stream(trajectory_filename.c_str());
-  carto::mapping::proto::Trajectory trajectory_proto;
-  CHECK(trajectory_proto.ParseFromIstream(&stream));
-
-  auto transform_interpolation_buffer =
-      carto::transform::TransformInterpolationBuffer::FromTrajectory(
-          trajectory_proto);
-
-  carto::io::PointsProcessorPipelineBuilder builder;
-  carto::io::RegisterBuiltInPointsProcessors(trajectory_proto, &builder);
-
   auto file_resolver =
       carto::common::make_unique<carto::common::ConfigurationFileResolver>(
           std::vector<string>{configuration_directory});
   const string code =
       file_resolver->GetFileContentOrDie(configuration_basename);
   carto::common::LuaParameterDictionary lua_parameter_dictionary(
       code, std::move(file_resolver));
-  const string tracking_frame =
-      lua_parameter_dictionary.GetString("tracking_frame");
 
+  std::ifstream stream(trajectory_filename.c_str());
+  carto::mapping::proto::Trajectory trajectory_proto;
+  CHECK(trajectory_proto.ParseFromIstream(&stream));
+
+  carto::io::PointsProcessorPipelineBuilder builder;
+  carto::io::RegisterBuiltInPointsProcessors(trajectory_proto, &builder);
   std::vector<std::unique_ptr<carto::io::PointsProcessor>> pipeline =
       builder.CreatePipeline(
           lua_parameter_dictionary.GetDictionary("pipeline").get());
 
-  tf2_ros::Buffer buffer;
-  // TODO(hrapp): Also parse tf messages and keep the 'buffer' updated as to
-  // support non-rigid sensor configurations.
+  auto tf_buffer = ::cartographer::common::make_unique<tf2_ros::Buffer>();
   if (!urdf_filename.empty()) {
```

Comment:
The assumption is that, if there's no URDF specified, then all the necessary transforms should be available on the /tf topic. Loading the URDF only makes sense for statically linked sensors (i.e. no movable joints). I fear combining the two approaches would only lead to headaches and I don't see a use case right now.